src/hooks/useInitializeProfileData.js

```diff
@@ -0,0 +1,30 @@
+import { captureException } from '@sentry/react-native';
+import { useCallback } from 'react';
+import { InteractionManager } from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+import { transactionSignaturesLoadState } from '@rainbow-me/redux/transactionSignatures';
+import logger from 'logger';
+
+export default function useInitializeProfileData() {
+  const dispatch = useDispatch();
+  const walletReady = useSelector(
+    ({ appState: { walletReady } }) => walletReady
+  );
+
+  const initializeProfileData = useCallback(async () => {
+    if (!walletReady) {
+      return;
+    }
+    try {
+      InteractionManager.runAfterInteractions(() => {
+        // Other profile related actions should be triggered here
+        dispatch(transactionSignaturesLoadState());
```

Comment:
We want to make sure to have these loaded for sure before fresh transactions are parsed